volatility3/framework/plugins/linux/proc.py

```diff
@@ -35,16 +41,144 @@ def get_requirements(cls):
                 element_type=int,
                 optional=True,
             ),
+            requirements.BooleanRequirement(
+                name="dump",
+                description="Extract listed memory segments",
+                default=False,
+                optional=True,
+            ),
+            requirements.ListRequirement(
+                name="address",
+                description="Process virtual memory addresses to include "
+                "(all other VMA sections are excluded). This can be any "
+                "virtual address within the VMA section.",
+                element_type=int,
+                optional=True,
+            ),
+            requirements.IntRequirement(
+                name="maxsize",
+                description="Maximum size for dumped VMA sections "
+                "(all the bigger sections will be ignored)",
+                default=cls.MAXSIZE_DEFAULT,
+                optional=True,
+            ),
         ]
 
+    @classmethod
+    def list_vmas(
+        cls,
+        task: interfaces.objects.ObjectInterface,
+        filter_func: Callable[
+            [interfaces.objects.ObjectInterface], bool
+        ] = lambda _: True,
+    ) -> Generator[interfaces.objects.ObjectInterface, None, None]:
+        """Lists the Virtual Memory Areas of a specific process.
+
+        Args:
+            task: task object from which to list the vma
+            filter_func: Function to take a vma and return False if it should be filtered out
+
+        Returns:
+            Yields vmas based on the task and filtered based on the filter function
+        """
+        if task.mm:
+            for vma in task.mm.get_mmap_iter():
+                if filter_func(vma):
+                    yield vma
+                else:
+                    vollog.debug(
+                        f"Excluded vma at offset {vma.vol.offset:#x} for pid {task.pid} due to filter_func"
+                    )
+
+    @classmethod
+    def vma_dump(
+        cls,
+        context: interfaces.context.ContextInterface,
+        task: interfaces.objects.ObjectInterface,
+        vm_start: int,
```

Comment:
Thanks, I know the method signature is a bit longer, but it's much easier for arbitrary callers of the classmethod to do so now.

Reply:
I was actually thinking of a making a generic write to file function as a utility somewhere. 

You either pass it a task object and it works off a new layer for that task, or you pass a layer name and it outputs that. 

Sending an addr and a length so it matches the layer read functions. 

What do you think? Not sure where exactly to put it if I did make it, Linux utilities?

Reply:
I'm not sure how common a task that would be?  If we only really do it in one plugin, then it might as well live in that plugin, but as a classmethod so other plugins can use it.  I think I'd prefer it to be in a layer rather than a task (since that's slightly more generic) and accept a list of tuples of ranges that should be concatenated together from the layer?  That could then live in just about any plugin at that point, but again, I just don't know how common it is to need that functionality...

Reply:
That makes sense a lot of sense. I was thinking about it for adding --dump to the pslist plugin but looks like @cstation has that covered. 

I'll leave it for now.